model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -39,6 +39,9 @@ async def generate(request: Request) -> Response:
     - stream: whether to stream the results or not.
     - other fields: the sampling parameters (See `SamplingParams` for details).
     """
+    # check health before accepting request and fail fast if engine isn't healthy
+    await engine.check_health()
```

Comment:
does this exit the pod if it fails? or does it just fail the request

Reply:
yeah i'm still testing this out, but this will just fail the request. I tried exit 1 as well but i don't think uvicorn itself exits if the underlying fastapi server exits so trying to fgiure out a solution for this

Reply:
mb, it might actually be ok to just exit the process? as long as whatever is holding the gpu memory gets killed I think it ends up being ok